### ISSUE TITLE
Can't Understand

### ISSUE THREAD

**Issue by HsiangNianian** [Possseidon/lua-midi#1]
So what do decimals in the output mean? Pitch?![Screenshot_2022-08-05-10-28-18-93_9e8df3d0c7c1f50248b6ee043a653d26.jpg](https://user-images.githubusercontent.com/44714368/182989278-98e81dcb-a622-4535-a1b2-ecaa705133f5.jpg)

**Comment by Possseidon** [comment]
Check out [examples/4-event-signatures.lua](https://github.com/Possseidon/lua-midi/blob/main/examples/4-event-signatures.lua). It's not 100% complete, but has descriptions (or at least sufficiently meaningful parameter names) for the basic midi events.

For your question in particular, `noteOn` has three parameters, namely `channel`, `key` and `velocity`.

- `channel` is a number between 0 and 15.
- `key` is the pitch (always an integer, that's just how midi works). You can check out [this table](https://www.music.mcgill.ca/~ich/classes/mumt306/StandardMIDIfileformat.html#BMA1_3) for what they mean.
- `velocity` the one you weren't sure about. `velocity` here means how loud the note is.
  Basically `0.0` means the same as `noteOff` (and is often used interchangeably) and `1.0` means maximum volume.

Note that `noteOff` also has a `velocity` parameter, which will pretty much always be `0.0`.

You can also take a look at [this table](https://www.music.mcgill.ca/~ich/classes/mumt306/StandardMIDIfileformat.html#BMA1_1) that lists all the different midi events.

**Comment by HsiangNianian** [comment]
> Check out [examples/4-event-signatures.lua](https://github.com/Possseidon/lua-midi/blob/main/examples/4-event-signatures.lua). It's not 100% complete, but has descriptions (or at least sufficiently meaningful parameter names) for the basic midi events.
> 
> For your question in particular, `noteOn` has three parameters, namely `channel`, `key` and `velocity`.
> 
> - `channel` is a number between 0 and 15.
> - `key` is the pitch (always an integer, that's just how midi works). You can check out [this table](https://www.music.mcgill.ca/~ich/classes/mumt306/StandardMIDIfileformat.html#BMA1_3) for what they mean.
> - `velocity` the one you weren't sure about. `velocity` here means how loud the note is.
>   Basically `0.0` means the same as `noteOff` (and is often used interchangeably) and `1.0` means maximum volume.
> 
> Note that `noteOff` also has a `velocity` parameter, which will pretty much always be `0.0`.
> 
> You can also take a look at [this table](https://www.music.mcgill.ca/~ich/classes/mumt306/StandardMIDIfileformat.html#BMA1_1) that lists all the different midi events.

我懂了！谢谢你！
